### ISSUE TITLE
False Negative for UnnecessaryParentheses Check

### ISSUE THREAD

**Issue by Kevin222004** [checkstyle/checkstyle#12961]
I have read check documentation: https://checkstyle.org/config_coding.html#UnnecessaryParentheses
I have downloaded the latest checkstyle 
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
cat Test.java
public class Test {

     public void method() {
          boolean a = true;
          int b = 42;
          int c = 42;
          int d = 32;
          if ((b<=c)!=a==(c>=d)) {
            return;
        }
     }
}

cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module
 PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="UnnecessaryParentheses" />
  </module>
</module>

kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Downloads/checkstyle-10.9.3-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
-------------------------------------

I am expecting Violation shown below
```
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test.java:8:15: Unnecessary parentheses around expression. [UnnecessaryParentheses]
[ERROR] /home/kevin/Desktop/check_style/Test.java:8:26: Unnecessary parentheses around expression. [UnnecessaryParentheses]
Audit done.
Checkstyle ends with 2 errors.
```

**Comment by romani** [comment]
Is it already described at https://checkstyle.org/config_coding.html#UnnecessaryParentheses_Notes as stuff that we can not support ?

@nrmancuso

**Comment by Kevin222004** [comment]
Yes I also missed that

**Comment by nrmancuso** [comment]
> Is it already described at https://checkstyle.org/config_coding.html#UnnecessaryParentheses_Notes as stuff that we can not support ?
> 

Yes, I missed this. @Kevin222004 let's close this issue if you agree with @romani

**Comment by Kevin222004** [comment]
Yes, I am closing this issue as it is clearly mentioned in doc that we can't support this
